agentstack/deploy.py

```diff
@@ -0,0 +1,94 @@
+import os
+import tempfile
+import tomllib
+import webbrowser
+import zipfile
+from pathlib import Path
+
+from agentstack.auth import get_stored_token, login
+from agentstack.conf import ConfigFile
+from agentstack.utils import term_color
+import requests
+
+
+def deploy():
+    bearer_token = get_stored_token()
+    if not bearer_token:
+        success = login()
+        if success:
+            bearer_token = get_stored_token()
+        else:
+            print(term_color("Failed to authenticate with AgentStack.sh", "red"))
+            return
+
+    project_id = get_project_id()
+    pyproject = load_pyproject()
+    files = list(Path('.').rglob('*.py'))
+
+    with tempfile.NamedTemporaryFile(suffix='.zip') as tmp:
+        with zipfile.ZipFile(tmp.name, 'w') as zf:
```

Comment:
Might be worth just building the project into a `whl` (which is essentially a zip file) since we already have a pyproject.toml file available in user projects (`pip build`). `pip install user-project.whl` after uploading it will install dependencies, too. 

https://packaging.python.org/en/latest/tutorials/packaging-projects/#generating-distribution-archives